Add F12 Quit window.

fmatthew5876 opened the pull request:
When you hold F12 a window appears and counts down. Once
countdown is complete, we reset to title screen.

Replaces 2k3e Alt+F12 feature. Enabled for all versions of the engine.

Also fixes a crash with F12, we capture old_scene before
checking for reset.

Ghabry commented on src/game_quit.h:
```diff
@@ -0,0 +1,47 @@
+
+#include "window_help.h"
+
+class Game_Quit {
```
Indent for public/private must be one level to the left

Ghabry commented on src/game_quit.h:
```diff
@@ -0,0 +1,47 @@
+		void Enable(bool value);
+	private:
+		void Reset();
+	private:
```
private twice

fmatthew5876 commented:
> Also makes now Steam more happy, as F12 is the screenshot key.

Except now anytime you try to create a stream screenshot you'll have this window

> An observable difference is that reset is now not working on Title and Load (from title) anymore but that's fine for me.

I was back and forth on this. But since it's supported in 2k3 I added it back. Also the title load menu kind of makes sense.

I also verified it does not appear in game browser.

> How about reset 2 seconds? 3 seconds feels a bit long and when TestPlay is enabled maybe instant as I use reset alot when testing and this breaks my Workflow https://xkcd.com/1172/ ;)

Changed to 2 seconds

Ghabry commented:
> Except now anytime you try to create a stream screenshot you'll have this window

There is a short delay after F12 until Steam creates a screenshot, so this should work.

carstene1ns commented on src/window_base.h:
```diff
@@ -40,7 +40,7 @@ class Window_Base : public Window {
 	 * @param width window width.
 	 * @param height window height.
 	 */
```
dox.

fmatthew5876 replied:
Fixed

fmatthew5876 commented:
I see what you mean about it being handy to not have this feature be too slow. In later PRs I changed it so fast forward affects the speed of the F12 window.

Ghabry commented:
I just noticed what you wrote. Which PR made it affected by FF?

fmatthew5876 commented:
#2017 the main loop refactor makes keeping this out of ff tricky. Also, after using it a lot now I think it's better that it does get sped up. It's still not too fast to make a mistake and makes testing easier.

Ghabry commented:
yeah I WANT it in FF because I'm already annoyed xD

carstene1ns commented:
Good thing we mostly run in test play mode 🤣
